### PR TITLE
[ci] Premainnet

### PR DESCRIPTION
**Pull request by rexhoffman**
## Motivation

Detect when premainnet can't handle transactions by running cluster test.

### Have you read the [Contributing Guidelines on pull requests]

Y

## Test Plan

Prior commits, only thing not tested is the circle integration since envs aren't sent to prs.

https://app.circleci.com/pipelines/github/libra/libra/18815/workflows/1a990185-3ef5-43ac-9753-28055727cd0b/jobs/108506/steps

## Related PRs

None

### CODE REVIEW

**Comment by rexhoffman** [comment]
Would like to use this to read from the google spreadsheet.

https://towardsdatascience.com/accessing-google-spreadsheet-data-using-python-90a5bc214fd2

but need permissions to add a project to fb.com google account.....

File: .circleci/config.yml [hunk 1]
```diff
@@ -617,3 +668,13 @@ workflows:
     jobs:
       - audit
       - code-coverage
+
+  premainnet-clustertest:
+    triggers:
+      - schedule:
+          cron: "0 * * * *"
```
Comment (rexhoffman) [inline .circleci/config.yml:675]: once an hour, on the hour.

File: .circleci/config.yml [hunk 2]
```diff
@@ -553,26 +557,73 @@ jobs:
...
-        filters:
-          branches:
-            ignore: gh-pages
-      - prefetch-crates:
-        filters:
-          branches:
-            ignore: gh-pages
+      - build-docker
+      - validate-cluster-test-dockerfile
+      - terraform
```
Comment (sausagee) [inline .circleci/config.yml:625]: Do we need these 3 jobs to run when building docs?

File: .circleci/config.yml [hunk 3]
```diff
@@ -553,26 +557,73 @@ jobs:
           name: Deploy to gh-pages branch
           command: |
             gh-pages --dotfiles --message "[skip ci] documentation update" --dist target/doc
-
+  premainnet_cluster_test:
+    executor: premainnet-cluster-test-executor
+    description: Run cluster test against Premainnet
+    steps:
+      - run:
+          name: Set Premainnet Clustetest Environment
```
[the remark this answers is not in the document]
Comment (rexhoffman) [reply .circleci/config.yml:565]: Sure, but the cred came from another source file (in this repo), link in the comment.  :D

File: .circleci/config.yml (hunk 2 above)
Comment (rexhoffman) [reply .circleci/config.yml:625]: the filter had no effect.   we need to put a small circle ci file with an ignore all in the gh-pages if we don't want circle ci to try to spin up (and immediately error out) on that branch.

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (rexhoffman) [reply .circleci/config.yml:565]: Done

File: .circleci/config.yml [hunk 4]
```diff
@@ -553,26 +557,71 @@ jobs:
...
+            cargo build -p cluster-test
+      - run:
+          name: Generate premaintnet config key
+          command: |
+            #premainnet seed found here: https://github.com/libra/partners/blob/master/terraform/validator/package.sh#L13
+            #premainnet number of nodes found here (login required)
+            #https://github.com/libra/partners/blob/master/helm/libra-validator/values.yaml#L2
+            cargo run -p config-builder -- faucet \
+            --output-dir "/tmp/config" \
+            --validators-in-genesis $PREMAINNET_NUM_VAL \
```
Comment (metajack) [inline .circleci/config.yml:608]: Are tjese env vars set somewhere?

Comment (rexhoffman) [reply .circleci/config.yml:608]: @metajack  yep, in the circleci project settings

File: .circleci/config.yml [hunk 5]
```diff
@@ -553,26 +557,72 @@ jobs:
...
             gh-pages --dotfiles --message "[skip ci] documentation update" --dist target/doc
-
+  premainnet-cluster-test:
+    executor: premainnet-cluster-test-executor
+    description: Run cluster test against Premainnet
+    steps:
+      - run:
+          name: Set Premainnet Clustetest Environment
+          command: |
+            # git revision in premainnet can be found on this site https://fburl.com/cynwp2sv:
```
Comment (sausagee) [inline .circleci/config.yml:567]: I'd suggest remove it here. Only those on intern can access.  In its place, this workflow can be referenced in the run book we run the perf test on pre-mainnet.

File: .circleci/config.yml [hunk 6]
```diff
@@ -553,26 +557,72 @@ jobs:
...
+            echo 'export GIT_REV_PREMIAN=163e20a' >> $BASH_ENV
+            # ip address to connect to can be found here (login required)
+            # https://docs.google.com/spreadsheets/d/1_6FUtGU1zdNcpl8-rRHgTH4dhrxGMdmfPfiC1KQHKaI?usp=sharing
```
Comment (sausagee) [inline .circleci/config.yml:570]: Would also suggest removing it here and relocating it to the run book.

File: .circleci/config.yml [hunk 7]
```diff
@@ -553,26 +557,72 @@ jobs:
...
-
+  premainnet-cluster-test:
+    executor: premainnet-cluster-test-executor
+    description: Run cluster test against Premainnet
+    steps:
+      - run:
+          name: Set Premainnet Clustetest Environment
+          command: |
+            # git revision in premainnet can be found on this site https://fburl.com/cynwp2sv:
+            echo 'export GIT_REV_PREMIAN=163e20a' >> $BASH_ENV
```
[the remark this answers is not in the document]
Comment (rexhoffman) [reply .circleci/config.yml]: Done, and moved to a slack environment variable to let folks update it easily.

File: .circleci/config.yml (hunk 5 above)
Comment (rexhoffman) [reply .circleci/config.yml]: Runbook has been updated and referenced.

**Comment by rexhoffman** [comment]
@sausagee since moving the git revision to an environment variable means we can't easily test this as pr, wanted to leave a second commit so you can reason about the change easily.

**Pull request closed without merging**